mapserver.h

```diff
@@ -2091,7 +2091,7 @@ void msPopulateTextSymbolForLabelAndString(textSymbolObj *ts, labelObj *l, char
   /**
   Sets up string-based mapfile loading and calls loadMapInternal to do the work
   */
-  MS_DLL_EXPORT mapObj *msLoadMapFromString(char *buffer, char *new_mappath);
+  MS_DLL_EXPORT mapObj *msLoadMapFromString(char *buffer, char *new_mappath, const configObj* config);
```

Comment:
@geographika I'm not sure it is appropriate to modify the signature of an exported function in the stable branch.
Creating a msLoadMapFromStringAndConfig() or something like that in the stable branch could be more appropriate

Reply:
@rouault - would you recommend updating the main branch to also use `msLoadMapFromStringAndConfig`?

Reply:
> would you recommend updating the main branch to also use `msLoadMapFromStringAndConfig`?

would probably be better

Reply:
To note - `msLoadMapFromString` is only exposed via MapScript (and only used by Python and C#).

Reply:
> To note - `msLoadMapFromString` is only exposed via MapScript

how do you know that it is only exposed via MapScript ?
e.g Ubuntu has a libmapserver-dev package with all mapserver headers and libmapserver.so (cf https://packages.ubuntu.com/bionic/amd64/libmapserver-dev/filelist), so potentially someone could embed it in their own application. I don't know if that's used in practice though

Reply:
Seems like an oversight from the original config implementation - we did change the msLoadMap() signature already. I'd think the risk is low but am fine with the additional function.

Reply:
Looking further, adding a `msLoadMapFromStringAndConfig` is going to duplicate most of the `msLoadMapFromString` function, diverge further from the equivalent `msLoadMap` and will also require changes to the MapScript implementations. I'm fine with not backporting - in this case would this update only appear in the 8.2 release? Were we thinking of dropping backports to the same branch anyway?

Reply:
Dropping backports from say 8.2 to 8.0 is what I'd favor. Maybe this is worth asking on mapserver-dev? To me it's something that should of been in the 8.0 and we just missed it. If it was there we would have just changed the signature and documented it - like was done with msLoadMap().

Reply:
Ah funny, maybe by taking the time to write what I was thinking, and re-reading my words, I can now see what you mean: I think you mean:
- no more use of backport
- simply when an 8.0.1 release is needed, for example, `main` would be tagged as `rel-8-0-1`

It sure is easy and clean to use the backport tool, it's been really great to use.

To be honest, I'd prefer to keep our branch use and backporting.

Reply:
> To be honest, I'd prefer to keep our branch use and backporting.

I also concur with that. ``main`` branch is 8.2dev and ``branch-8-0`` for 8.0.x bugfix releases. That's the quite common pattern among other projects I'm involved with.

Reply:
I just want to refrain from backporting to previous branches in the same major release. So once 8.2 is out, main is 8.4-dev and the 8.0 branch gets no further updates.